### PR TITLE
feat(StatusTagSelector): Replaced member delegate w/ StatusMemberLisItm

### PR DESCRIPTION
**Pull request by alexandraB99**
Fixes https://github.com/status-im/status-desktop/issues/5625

### Checklist

- [x] follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
  - the scope should be the component's name e.g: `feat(StatusListItem): ... `
  - when adding new components, the scope is the module e.g: `feat(StatusQ.Controls): ...`
- [x] add documentation if necessary (new component, new feature)
- [x] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [x] test changes in both light and dark theme?
- [ ] is this a breaking change?
    - [ ] use the dedicated `BREAKING CHANGE` commit message section
    - [ ] resolve breaking changes in [status-desktop](https://github.com/status-im/status-desktop)
        - [ ] (pre-merge) adapt code to breaking changes
        - [ ] (post-merge) update StatusQ submodule pointer
- [x] test changes in [status-desktop](https://github.com/status-im/status-desktop)

### CODE REVIEW

**Comment by alexandraB99** [comment]
Please note that highlight doesn't show because of screen recording stealing mouse focus.

https://user-images.githubusercontent.com/31625338/168632793-a1291ac4-ca42-4763-846d-3743b06699e9.mov

File: sandbox/demoapp/data/Models.qml [hunk 1]
```diff
@@ -896,12 +844,12 @@ CExPynn1gWf9bx498P7/nzPcxEzGExhBdJGYihtAYQlO+tUZvqrPbqeudo5iJGEJjCE15a3VtodH3q2I
         id: membersList
         ListElement {
             nickName: "This is an example"
-            userName: "annabelle"
-            chatKey: "0x043a7ed0e8752236a4688563652fd0296453cef00a5dcddbe252dc74f72cc1caa97a2b65e4a1a52d9c30a84c9966beaaaf6b333d659cbdd2e486b443ed1012cf04"
+            name: "Maria"
+            publicId: "0x043a7ed0e8752236a4688563652fd0296453cef00a5dcddbe252dc74f72cc1caa97a2b65e4a1a52d9c30a84c9966beaaaf6b333d659cbdd2e486b443ed1012cf04"
```
Comment (osmaczko) [inline sandbox/demoapp/data/Models.qml:848]:
```suggestion
            pubKey: "0x043a7ed0e8752236a4688563652fd0296453cef00a5dcddbe252dc74f72cc1caa97a2b65e4a1a52d9c30a84c9966beaaaf6b333d659cbdd2e486b443ed1012cf04"
```

**Review by osmaczko — APPROVED** [review]

**Review by noeliaSD — COMMENTED** [review]
It looks nice!! Just a comment.. It should be set the color of the idle background that rounds the connect / disconnect circle in order to have the expected behavior. Maybe it needs to be fixed in `StatusMemberListItem` or maybe it directly inculdes a property to set it (not checked).
See the video for better understanding (first part of the video, expected behavior):

https://user-images.githubusercontent.com/97019400/168742503-4dc93bb5-6053-47d4-824b-caa7e3c30a8a.mov

File: sandbox/demoapp/data/Models.qml (hunk 1 above)
Comment (alexandraB99) [reply sandbox/demoapp/data/Models.qml:848]: @osmaczko I used the naming of the real model, which is `publicId`

File: sandbox/demoapp/data/Models.qml [hunk 2]
```diff
@@ -913,23 +861,23 @@ CExPynn1gWf9bx498P7/nzPcxEzGExhBdJGYihtAYQlO+tUZvqrPbqeudo5iJGEJjCE15a3VtodH3q2I
...
-            userName: ""
-            chatKey: ""
+            nickName: ""
+            name: "carmen.eth"
+            publicId: "0x043a7ed78362567894688563652fd0296453cef00a5dcddbe252dc74f72cc1caa97a2b65e4a1a52d9c30a84c9966beaaaf6b333d659cbdd2e486b443ed1012cf04"
             trustIndicator: StatusContactVerificationIcons.TrustedType.Untrustworthy
             isMutualContact: false
-            isOnline: false
-            source: ""
+            onlineStatus: false
```
[the remark this answers is not in the document]
Comment (alexandraB99) [reply sandbox/demoapp/data/Models.qml:869]: same here `onlineStatus` is how it's named in the real model, we have to use the real names as this component is basically tight to the real model (used in both sandbox and desktop app), so we cannot use different names. this model here is just for the sandbox app to run

File: src/StatusQ/Components/StatusTagSelector.qml [hunk 3]
```diff
@@ -98,6 +98,14 @@ Item {
         This property sets the tags count limit.
     */
     property int nameCountLimit: 5
+
+    /*!
+        \qmlproperty var StatusTagSelector::ringSpecModel
+        This property holds the function to calculate the ring spec model
+        based on the public key.
+    */
+    property var ringSpecModel
```
[the remark this answers is not in the document]
Comment (alexandraB99) [reply src/StatusQ/Components/StatusTagSelector.qml:107]: what do you suggest? maybe `return []`;

**Comment by alexandraB99** [comment]
> It looks nice!! Just a comment.. It should be set the color of the idle background that rounds the connect / disconnect circle in order to have the expected behavior. Maybe it needs to be fixed in `StatusMemberListItem` or maybe it directly inculdes a property to set it (not checked). See the video for better understanding (first part of the video, expected behavior):
> 
>  Screen.Recording.2022-05-17.at.08.10.39.mov

@noeliaSD fixed

File: src/StatusQ/Components/StatusTagSelector.qml (hunk 3 above)
[the remark this answers is not in the document]
Comment (alexandraB99) [reply src/StatusQ/Components/StatusTagSelector.qml:107]: done!

File: sandbox/demoapp/data/Models.qml (hunk 2 above)
[the remark this answers is not in the document]
Comment (osmaczko) [reply sandbox/demoapp/data/Models.qml:869]: Got it, tight coupling with desktop kicks in again :man_shrugging:

**Review by noeliaSD — APPROVED** [review]

**Pull request merged**